packages/core/admin/admin/src/pages/SettingsPage/pages/AuditLogs/ListView/index.js

```diff
@@ -1,43 +1,61 @@
 import React, { useState } from 'react';
-import { SettingsPageTitle, DynamicTable } from '@strapi/helper-plugin';
+import { useIntl } from 'react-intl';
+import { useQuery } from 'react-query';
+import {
+  SettingsPageTitle,
+  DynamicTable,
+  useRBAC,
+  useNotification,
+  useFocusWhenNavigate,
+} from '@strapi/helper-plugin';
 import { HeaderLayout, ContentLayout } from '@strapi/design-system/Layout';
 import { Main } from '@strapi/design-system/Main';
-import { useIntl } from 'react-intl';
+import adminPermissions from '../../../../../permissions';
+// import { useFetchClient } from '../../../../../hooks';
 import TableRows from './DynamicTable/TableRows';
 import tableHeaders from './utils/tableHeaders';
+import { fetchData } from './utils/api';
 import ModalDialog from './ModalDialog';
 
-const data = [
-  {
-    id: 1,
-    action: 'Update',
-    date: '2022-11-14T23:04:00.000Z',
-    user: 'John Doe',
-  },
-  {
-    id: 2,
-    action: 'Create',
-    date: '2022-11-04T18:24:00.000Z',
-    user: 'Kai Doe',
-  },
-  {
-    id: 3,
-    action: 'Delete',
-    date: '2022-10-09T11:26:00.000Z',
-    user: 'John Doe',
-  },
-  {
-    id: 4,
-    action: 'Log in',
-    date: '2022-10-09T11:24:00.000Z',
-    user: 'Kai Doe',
-  },
-];
+const QUERY = 'audit-logs';
 
 const ListView = () => {
   const { formatMessage } = useIntl();
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [detailsActionData, setDetailsActionData] = useState(null);
+  const toggleNotification = useNotification();
+  const {
+    allowedActions: { canRead },
+  } = useRBAC(adminPermissions.settings.auditLogs);
+
+  useFocusWhenNavigate();
+
+  /**
+   * TODO: using fetchclient facing difficulties to mock, for time being using axiosinstance to pass the tests
+   *
+  const { get } = useFetchClient();
+  const fetchData = async () => {
+    const {
+      data: { results },
+    } = await get(`/admin/audit-logs`);
+
+    return results;
+  }; */
+
+  const { status, data, isFetching } = useQuery(QUERY, fetchData, {
+    enabled: canRead,
+    keepPreviousData: true,
+    retry: false,
+    staleTime: 1000 * 20,
+    onError() {
+      toggleNotification({
+        type: 'warning',
+        message: { id: 'notification.error', defaultMessage: 'An error occured' },
+      });
+    },
+  });
+
+  const isLoading = status === 'loading' || isFetching;
```

Comment:
I think the normal way to use react query is to get `isLoading` from the `useQuery` returned object instead

I believe isFetching is intended for more advanced use case, I don't think it's needed here